### PR TITLE
Clean up build id logic.

### PR DESCRIPTION
**Pull request by briantting**
b/294867970

### CODE REVIEW

File: cobalt/build/BUILD.gn [hunk 1]
```diff
@@ -13,15 +13,9 @@
...
-  cobalt_version = exec_script("get_build_id.py", [], "trim string")
-
   script = "build_id.py"
   outputs = [ "$root_gen_dir/cobalt_build_id.h" ]
-
-  args = [
-    rebase_path(outputs[0], root_build_dir),
-    cobalt_version,
-  ]
+  args = [ rebase_path(outputs[0], root_build_dir) ]
```
Comment (andrewsavage1) [inline cobalt/build/BUILD.gn:18]: Note that this change changes the time of build id generation from the GN step to the Ninja step, which can have different results. afaik it won't affect anything, but something to keep in mind

**Review by oxve — APPROVED** [review]
If Andrew's comment isn't a concern I don't see any issues with this.

File: cobalt/build/BUILD.gn (hunk 1 above)
Comment (briantting) [reply cobalt/build/BUILD.gn:18]: That should be fine. I just doubled checked this, but it seems that this isn't the case. When I run the gn step neither the cobalt_build_id.h nor the build_info.json files are generated until the ninja step.

Comment (andrewsavage1) [reply cobalt/build/BUILD.gn:18]: It is the case—while the cobalt_build_id.h file isn't yet generated after running GN, the cobalt_version is already set. The preset cobalt_version is written to cobalt_build_id.h at compilation (ninja) time.

Comment (briantting) [reply cobalt/build/BUILD.gn:18]: Ah! Gotcha, though I think it being set and generated at the same step now is easier to follow. Either way I don't see us running into any issues with it being set 1 step later now.

**Pull request merged**